util/slack_wrapper.py

```diff
@@ -1,202 +1,216 @@
         """
         Rename an existing channel.
         """
-        api_call = "groups.rename" if is_private else "channels.rename"
 
-        return self.client.api_call(api_call, channel=channel_id, name=new_name, validate=False)
+        return self.client.conversations_rename(channel=channel_id, name=new_name).data
 
-    def get_channel_info(self, channel_id, is_private=False):
+    def get_channel_info(self, channel_id: str) -> Dict:
         """
         Return the channel info of a given channel ID.
         """
 
-        api_call = "conversations.info"
-        return self.client.api_call(api_call, channel=channel_id)
+        return self.client.conversations_info(channel=channel_id).data
 
-    def get_channel_members(self, channel_id, next_cursor=None):
-        """Recursively fetch members of the given channel, until none remain to be fetched"""
-        response = self.client.api_call("conversations.members", channel=channel_id, cursor=next_cursor)
-        members = response['members']
-        next_cursor = response['response_metadata']['next_cursor']
-        if not next_cursor:
-            return members
-        else:
-            return members + self.get_channel_members(channel_id, next_cursor)
+    def get_channel_members(self, channel_id: str) -> List[str]:
+        """ Return list of member ids in a given channel ID. """
 
-    def update_channel_purpose_name(self, channel_id, new_name, is_private=False):
+        members = []
+        for page in self.client.conversations_members(channel=channel_id):
+            members.extend(page["members"])
+
+        return members
+
+    def update_channel_purpose_name(self, channel_id: str, new_name: str) -> Dict:
         """
         Updates the channel purpose 'name' field for a given channel ID.
         """
 
-        # Update channel purpose
-        channel_info = self.get_channel_info(channel_id, is_private)
+        channel_info = self.get_channel_info(channel_id)
 
-        if channel_info:
-            purpose = load_json(channel_info['channel']['purpose']['value'])
-            purpose['name'] = new_name
+        purpose = load_json(channel_info["channel"]["purpose"]["value"])
+        purpose["name"] = new_name
 
-            self.set_purpose(channel_id, json.dumps(purpose), is_private)
+        return self.set_purpose(channel_id, json.dumps(purpose))
 
-    def post_message(self, channel_id, text, timestamp="", parse="full"):
+    def post_message(self, channel_id: str, text: str, timestamp="", parse="full") -> Dict:
         """
         Post a message in a given channel.
         channel_id can also be a user_id for private messages.
         Add timestamp for replying to a specific message.
         """
-        self.client.api_call("chat.postMessage", channel=channel_id,
-                             text=text, as_user=True, parse=parse, thread_ts=timestamp)
+        if not isinstance(text, str):
```

Comment:
How does this happen? Can we root out any clients doing this with mypy?

Reply:
During testing I accidentally hit an exception and it tried to post that directly to slack. There were a little too many calls to `post_message` for me to dig through but we should definitely run mypy on this once we add type annotations to everything